packages/block-editor/src/hooks/duotone.js

```diff
@@ -290,25 +267,47 @@ function DuotoneStyles( {
 		// Assuming the selector part is a subclass selector (not a tag name)
 		// so we can prepend the filter id class. If we want to support elements
 		// such as `img` or namespaces, we'll need to add a case for that here.
-		return `.editor-styles-wrapper .${ filterId }${ selectorPart.trim() }`;
+		return `.${ filterId }${ selectorPart.trim() }`;
 	} );
 
 	const selector = selectorsScoped.join( ', ' );
 
 	const isValidFilter = Array.isArray( colors ) || colors === 'unset';
 
-	return (
-		element &&
-		isValidFilter &&
-		createPortal(
-			<InlineDuotone
-				selector={ selector }
-				id={ filterId }
-				colors={ colors }
-			/>,
-			element
-		)
+	const { setStyleOverride, deleteStyleOverride } = unlock(
+		useDispatch( blockEditorStore )
 	);
+
+	useEffect( () => {
+		if ( ! isValidFilter ) return;
+
+		setStyleOverride( filterId, {
+			css:
+				colors !== 'unset'
+					? getDuotoneStylesheet( selector, filterId )
+					: getDuotoneUnsetStylesheet( selector ),
+			__unstableType: 'presets',
+		} );
+		setStyleOverride( `duotone-${ filterId }`, {
+			assets:
+				colors !== 'unset' ? getDuotoneFilter( filterId, colors ) : '',
+			__unstableType: 'svgs',
+		} );
+
+		return () => {
+			deleteStyleOverride( filterId );
+			deleteStyleOverride( `duotone-${ filterId }` );
+		};
+	}, [
+		isValidFilter,
+		colors,
+		selector,
+		filterId,
+		setStyleOverride,
+		deleteStyleOverride,
+	] );
+
+	return null;
```

Comment:
Minor: If it's not a public Api, this `DuotoneStyles` component could probably be a `useDuotoneStyles` hook since it renders nothing.

Reply:
Well, that's exactly the whole point of this PR 👍